header_fix.py

```diff
@@ -0,0 +1,146 @@
+        if idx == -1:
+            new_content.append(line)
+            continue
+        needs_write = True
+        if header.startswith('../'):
+            copy_recur(os.path.join(file_dir, header), file_dir)
+            line = line.replace('../', '')
+        elif header.startswith('./'):
+            line = line.replace('./', '')
+        else:
+            # this is a big assumption: That is the header it tries to include here is in a subdir we already flattened
+            base = os.path.basename(header)
+            line = f'#include "{base}"'
+        line += ' // modified by Poolside fork'
+        new_content.append(line)
+    if needs_write:
+        with open(file_path, 'w') as file:
+            file.write('\n'.join(new_content))
+
+def add_extra_line(file_path, extra_line):
+    with open(file_path, 'r') as file:
+        content = file.read()
+    with open(file_path, 'w') as file:
+        file.write(extra_line + content + extra_line)
+
+# generate a comment header/footer to make it clear a file is modified by us, not directly from upstream
+def make_comments():
+    comment_line = '////////////////////////////////////////////////////////////////////////////////\n'
+    return f'\n\n{comment_line}// Poolside modification in the fork, generated at {datetime.datetime.now()} //\n{comment_line}\n\n'
+
+
+# move file from src to dst if src exists, overwrite dst if it exists.
+def force_move_file(src, dst):
+    # if src no longer exists, we proabably done moving it already. Abort:
+    if not os.path.exists(src):
+        print(f'File {src} no longer exists, not going to move anything.')
+        return
+    # if dst exists, we want to overwrite it
+    if os.path.exists(dst):
+        print(f'Warning: removing existing file {dst}')
+        os.remove(dst)
+    os.rename(src, dst)
+    add_extra_line(dst, make_comments())
+    print(f'File {src} has been moved to {dst}.')
+
+
+def endswith_any(filename, extenstions):
+    return any(filename.endswith(extension) for extension in extenstions)
+
+
+def op_recur(src, dst_dir, operation):
+    headers = []
+    with open(src, 'r') as f:
+        for lines in f:
+            if lines.startswith('#include') and lines.count('"') > 0:
+                headers.append(lines.split('"')[1])
+    for h in headers:
+        op_recur(h, dst_dir, operation)
+    operation(src, os.path.join(dst_dir, os.path.basename(src)))
+
+
+def copy_with_extra_commends(src, dst):
+    shutil.copy2(src, dst)
+    add_extra_line(dst, make_comments())
+
+def copy_recur(src, dst_dir):
+    op_recur(src, dst_dir, copy_with_extra_commends)
+    print(f'File {src} has been copied to {dst_dir}.')
+
+
+# (1) Fix up go-tree-sitter root level headers so they do not have #include "./foo.h"
+# (2) Flatten a list of language subdirs that we care about. No more go-tree-sitter/{lang}/foo/bar.h
+# (3) Then in these flatten subdirs, fix up any header inclusion that isn't directly a sibling file.
+if __name__ == '__main__':
```

Comment:
mind throwing a readme with instructions how and when this script should be used?

Reply:
added a new README_poolside.md file